chrome/content/zotero/xpcom/sync/syncRunner.js

```diff
@@ -230,6 +236,24 @@ Zotero.Sync.Runner_Module = function (options = {}) {
 		
 		return json;
 	});
+
+
+	// Prompt if library empty and there is no userID stored
+	this.checkEmptyLibrary = Zotero.Promise.coroutine(function* (client) {
+		let library = Zotero.Libraries.userLibrary;
+		let userID = Zotero.Users.getCurrentUserID();
+
+		if(!userID && !library.hasItems()) {
+			if(! Services.prompt.confirm(
+				null,
+				Zotero.getString('general.warning'),
+				Zotero.getString('sync.warning.emptyLibrary', Zotero.clientName)
+			)) {
+				return false;
+			}
+		}
```

Comment:
We generally avoid `confirm`, because it doesn't provide control over the button titles. Should use `confirmEx` with `sync.sync` and either `sync.cancel` or just "Cancel".